### DISCUSSION TITLE
How to get Test results

### DISCUSSION THREAD

**Discussion by sepsky** [danielpaulus/go-ios#254] [Q&A]
Hi, 

I've got a Runner app for a scheme containing all my UITests. I created the runner using xcodebuild (build-for-testing), which created the `xctestrunner` as well as the two apps (`XYZ.app` and `XYZ-runner.app`). 

I can run both of the apps using go-ios (I can't run the `xctestrunner`), but I'm not sure where we'd get the test results for each test case (inside the scheme for which I created the builds). 

I've been logging device console with `ios syslog...`, but due to sheer volume of reported lines, I was wondering if there's a better way of grabbing these test results? 

Thanks and if I can provide more info please let me know. 

Sep

**Comment by wellkilo** [comment]
Launching the app and runner is not the same as starting an XCTest session. Use ios runtest or ios runxctest; go-ios then connects to testmanagerd and collects per-suite/per-case callbacks.

With the generated .xctestrun:

    ios runxctest \
      --xctestrun-file-path /path/to/YourScheme_iphoneos.xctestrun \
      --log-output test.log \
      --junit-output results.xml

Or with installed bundle IDs:

    ios runtest \
      --bundle-id com.example.XYZ \
      --test-runner-bundle-id com.example.XYZUITests.xctrunner \
      --xctest-config XYZUITests.xctest \
      --log-output test.log \
      --junit-output results.xml

Use --log-output - to stream the test listener output to stdout. The JUnit file is the useful CI artifact: it contains suites, cases, failures/skips and durations, avoiding device-syslog filtering.

The current CLI documents these flags for both commands: [main.go](https://github.com/danielpaulus/go-ios/blob/c584042907dee35bf80377284ae3aac85990b4d7/main.go#L465-L482). The implementation serializes the collected TestSuite values through the JUnit writer in [cmd_device_testing.go](https://github.com/danielpaulus/go-ios/blob/c584042907dee35bf80377284ae3aac85990b4d7/cmd_device_testing.go#L16-L22).

If your installed build lacks --junit-output, update go-ios; this functionality was added after the original question.
